### ISSUE TITLE
[a11y]: Modal interaction problems

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#12727]
### Description
This issue covers what I would consider 'bugs' in the modal components. It is complementary to the discussion https://github.com/carbon-design-system/carbon/discussions/12726 I have opened, and contains information based on the Modal component review for accessibiliity. It is complementary to a review of Carbon website guidance https://github.com/carbon-design-system/carbon-website/issues/3312.

### Items to address
- [ ] https://github.com/carbon-design-system/carbon/issues/13710
- [ ] https://github.com/carbon-design-system/carbon/issues/13711
- [ ] https://github.com/carbon-design-system/carbon/issues/13680
- [ ] https://github.com/carbon-design-system/carbon/issues/13714

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-modal--with-state-manager

**Comment by mbgower** [comment]
I just encountered a 'modal' dialog in Workday. Obviously not a Carbon component, but a real-life situation helps discuss. (Note that View Details did not have focus when this appeared. That was a result of me tabbing.)

My keyboard was constrained to tabbing between the X button to close and the link to "View details".  Even though there was no decision needed on my part, I was forced to intentionally dismiss via keyboard.
However, with a mouse, I could simply click anywhere and the 'modal' disappeared.
In such a situation, it would be easy for the user to not even see the message before dismissing. For a dialog that required interaction, that would be a poor user experience. In this situation (where a response wasn't really needed) the impact is neglibile.
So, should it BE a modal?
And if it isn't a modal, should tabbing away from it dismiss it? Is there any dropback to that interaction?
![image](https://user-images.githubusercontent.com/14298245/205165444-75d5d389-9400-4455-976e-686bf5c9c5cc.png)

**Comment by mbgower** [comment]
@sstrubberg , IMO the items marked as Sev 2 remain appropriate for higher priority (issues 1, 2, 5 and 7.2)

**Comment by tay1orjones** [comment]
@sstrubberg We should break these out into individual issues and treat this as an umbrella.
